doc/rules/phpdoc/phpdoc_order.rst

```diff
@@ -28,10 +63,55 @@ Example #1
    + * @return int  Return the number of changes.
      */
 
+Example #3
+~~~~~~~~~~
+
+With configuration: ``['order' => ['param', 'return', 'throws']]``.
+
+.. code-block:: diff
+
+   --- Original
+   +++ New
+    <?php
+    /**
+     * Hello there!
+     *
+   - * @throws Exception|RuntimeException foo
+     * @custom Test!
+   - * @return int  Return the number of changes.
+     * @param string $foo
+     * @param bool   $bar Bar
+   + * @return int  Return the number of changes.
+   + * @throws Exception|RuntimeException foo
+     */
+
+Example #4
+~~~~~~~~~~
+
+With configuration: ``['order' => ['param', 'custom', 'throws', 'return']]``.
+
+.. code-block:: diff
+
+   --- Original
+   +++ New
+    <?php
+    /**
+     * Hello there!
+     *
+   + * @param string $foo
+   + * @param bool   $bar Bar
+   + * @custom Test!
+     * @throws Exception|RuntimeException foo
+   - * @custom Test!
+     * @return int  Return the number of changes.
+   - * @param string $foo
+   - * @param bool   $bar Bar
+     */
+
 Rule sets
 ---------
 
 The rule is part of the following rule set:
 
 @PhpCsFixer
-  Using the `@PhpCsFixer <./../../ruleSets/PhpCsFixer.rst>`_ rule set will enable the ``phpdoc_order`` rule.
+  Using the `@PhpCsFixer <./../../ruleSets/PhpCsFixer.rst>`_ rule set will enable the ``phpdoc_order`` rule with the default config.
```

Comment:
It should be added to the Symfony ruleset with the config:

```
@param, @return, @throws
```
according to
https://symfony.com/doc/current/contributing/code/standards.html

Reply:
Suggestion applied!

Reply:
Hey @VincentLanglet, I've got a test issue with your suggestion.

`RuleSetTest::testThatDefaultConfigIsNotPassed` throws an error:
`Rule "phpdoc_order" (in RuleSet "@ Symfony") has default config passed.`

It's all because the test **sorts** configuration arrays before comparing them.

Do you have any idea how to approach this problem?

For the record:
Default config for the fixer is: `['order' => ['param', 'throws', 'return']]`
@ Symfony's config for the fixer is: `['order' => ['param', 'return', 'throws']]`

Reply:
Since order has impact, this shouldn't be sorted.

Dunno if
- The sort should be removed
- An option should be added to not always sort for this test

but in your case this should definitively not be sorted.

Reply:
Ok, I did it this way:

```
[...]
        // do not sort options for which order is important
        $excludeFromSorting = $this->orderMattersFor($ruleName);

        static::assertNotSame(
            $this->sortNestedArray($defaultConfig, $excludeFromSorting),
            $this->sortNestedArray($ruleConfig, $excludeFromSorting),
            sprintf('Rule "%s" (in RuleSet "%s") has default config passed.', $ruleName, $setName)
        );

[...]
```

```
[...]
    private function orderMattersFor(string $ruleName): array
    {
        static $orderMattersFor = [
            'phpdoc_order' => ['order'],
        ];

        return $orderMattersFor[$ruleName] ?? [];
    }
[...]
```

```
[...]
    private function sortNestedArray(array $array, array $excludeKeys = []): array
    {
        foreach ($array as $key => $element) {
            if (!\is_array($element) || \in_array($key, $excludeKeys, true)) {
                continue;
            }

            $array[$key] = $this->sortNestedArray($element, $excludeKeys);
        }
[...]
```

Reply:
... but I think it's wrong :)

Reply:
> ``RuleSetTest::testThatDefaultConfigIsNotPassed`` throws an error:
> ``Rule "phpdoc_order" (in RuleSet "@Symfony") has default config passed.``
> 
> It's all because the test sorts configuration arrays before comparing them.

@SpacePossum or @jrmajor? Could you please shed some light on this matter? 
I'm stuck with it. I want to minimize code changes, but I am losing clarity of meaning.

For example, I can use the normalizer to convert ``array`` to ``ArrayIterator`` so that ``RuleSetTest`` will skip sorting the value of the ``order`` option. 
 
https://github.com/gdl2studio/PHP-CS-Fixer/blob/0650d49124ad7dc2347047cec9a90000d4b848dc/src/Fixer/Phpdoc/PhpdocOrderFixer.php#L111